infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/local/projects/BindMountProjectsVolumeProvisioner.java

```diff
@@ -65,24 +69,26 @@ public ProjectsVolumeProvisioner(
   public void provision(DockerEnvironment internalEnv, RuntimeIdentity identity)
       throws InfrastructureException {
 
-    Optional<String> devMachineOpt = WsAgentMachineFinderUtil.getWsAgentServerMachine(internalEnv);
-    if (!devMachineOpt.isPresent()) {
-      // should not happen
-      // no wsagent - no projects volume is needed
-      return;
-    }
-    String devMachineName = devMachineOpt.get();
+    for (Entry<String, DockerContainerConfig> containerEntry :
+        internalEnv.getContainers().entrySet()) {
 
-    DockerContainerConfig devMachineConfig = internalEnv.getContainers().get(devMachineName);
-    if (devMachineConfig == null) {
-      throw new InternalInfrastructureException(
-          format("Docker configuration for machine '%s' not found", devMachineName));
+      DockerContainerConfig value = containerEntry.getValue();
+      List<String> newVolumes = new ArrayList<>();
+      for (String volume : value.getVolumes()) {
+        String[] volumeSourceTarget = volume.split(":");
+        if (VolumeNames.matches(volumeSourceTarget[0], "projects", identity.getWorkspaceId())) {
```

Comment:
Maybe It'd be better to make `projects` as public constant of ProjectsVolumeForWsAgentProvisioner class

Reply:
ok

Reply:
done